### ISSUE TITLE
Can't import TXT file

### ISSUE THREAD

**Issue by Jetschoo** [jasp-stats/jasp-issues#2910]
### JASP Version

0.19.0

### JASP Module

Unrelated

### What OS are you seeing the problem on?

Windows 11

### Bug Description

Whenever I try to import a file it never opens. About halfway it will get stuck loading and no matter how long I let it run it won't open the file. I've tried uninstalling and installing JASP again, I've tried the repair function, I've tried different kind of TXT files and rebooting my device but nothing seems to work.
See the image for more clarification:
![20240912_200612](https://github.com/user-attachments/assets/b7d7142d-b9bb-4510-81fe-8913a4c22689)

### Expected Behaviour

It should've opened the file(s).

### Steps to Reproduce

1. Open jasp
2.select open
3. Select a file
4. the file never stops loading
...

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if possible.
- [ ] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by tomtomme** [comment]
Can you share the file?

**Comment by boutinb** [comment]
Do you have the save problem with csv file?
Maybe this is a permission issue.  Can you send us the log files: https://github.com/jasp-stats/jasp-desktop/blob/development/Docs/user-guide/logging-howto.md ?
If you install JASP with the pre-installed zip file, do you have the same issue?

**Comment by TiesGoedman** [comment]
Hi, i'm seemingly having the same problem. I tested it and the problem persists with both the msi installed version and the pre installed zip version. It also affects both importing from txt and csv.  I'll upload the logs down below + the file im trying to import.

I'm also on windows 11 and have JASP 0.19.1

Let me know if you need any other information!

[JASP 2024-09-14 14_13_06 Desktop.log](https://github.com/user-attachments/files/17002371/JASP.2024-09-14.14_13_06.Desktop.log)
[JASP 2024-09-14 14_13_06 Engine 0.log](https://github.com/user-attachments/files/17002372/JASP.2024-09-14.14_13_06.Engine.0.log)
[JASP 2024-09-14 14_13_58 Desktop.log](https://github.com/user-attachments/files/17002373/JASP.2024-09-14.14_13_58.Desktop.log)
[JASP 2024-09-14 14_13_58 Engine 0.log](https://github.com/user-attachments/files/17002374/JASP.2024-09-14.14_13_58.Engin
[lesson_02_opdracht1-2_ensembl_GRCh38.txt](https://github.com/user-attachments/files/17002375/lesson_02_opdracht1-2_ensembl_GRCh38.txt)
e.0.log)

**Comment by JorisGoosen** [comment]
This is a rather big file for JASP at the moment as it contains a lot of strings.
But I think I can perhaps find some more optimizations.

**Comment by JorisGoosen** [comment]
It was good I looked at this!

I think Ive managed to speed up the reading process with lots of strings quite a bit.
The file linked above now loads in 1 minute on my humble laptop.

**Comment by TiesGoedman** [comment]
Thankyou! It works for me now!

**Comment by JorisGoosen** [comment]
Very nice, keep in mind that code is rather experimental atm though!

**Comment by boutinb** [comment]
There are 2 columns in your dataset that slow down a lot the loading in JASP:  `Gene stable ID` and `Gene name`. As @JorisGoosen this should be solved in the next JASP version.
But for the time being I have made a version of your datafile by converting the `Gene stable ID` column into a column with only integers, and I just removed the gene name. This file loads in few seconds on my laptop. I'm not sure if this file is usable for you, but in case I upload it here.
[lesson_with_geneid.csv](https://github.com/user-attachments/files/17097495/lesson_with_geneid.csv)
